src/App/Fossa/VPS/Scan.hs

```diff
@@ -43,95 +39,22 @@ scanMain basedir apiOpts metadata logSeverity overrideProject fileFilters skipIp
-  SherlockInfo{..} <- getSherlockInfo apiOpts
-
-  -- Build locator info
-  let locator = createLocator projectName sherlockOrgId
-  let revisionLocator = createRevisionLocator projectName sherlockOrgId projectRevision
-
-  -- FIXME: use better type here
-  let vpsOpts = VPSOpts projectName (Just projectRevision) (fromFlag SkipIPRScan skipIprFlag) fileFilters
-
-  -- Update vpsOpts with overriding scan filter blob. 
-  -- Previous uses of `vpsOpts` do not deconstruct the object due to this not yet being overridden.
-  (vpsOpts'@VPSOpts{..}, areFiltersOverridden) <- overrideScanFilters apiOpts vpsOpts locator
-  logDebug $ pretty $ "[All] Creating project with locator '" <> unRevisionLocator revisionLocator <> "'"
-
-  -- Create scan in Core
-  logDebug "[All] Creating project in FOSSA"
-  _ <- context "creating project in FOSSA" $ createCoreProject vpsProjectName projectRevision metadata apiOpts
-
-  -- Create scan in SY
-  logDebug "[All] Creating scan in Scotland Yard"
-  let syOpts = ScotlandYardOpts locator projectRevision sherlockOrgId vpsOpts'
-  response <- context "creating scan ID" $ createScotlandYardScan apiOpts syOpts
-  let scanId = createScanResponseId response
+  ) => BaseDir -> Severity -> OverrideProject -> Flag SkipIPRScan -> Flag LicenseOnlyScan -> FilterExpressions -> ApiOpts -> ProjectMetadata -> BinaryPaths -> m ()
+vpsScan (BaseDir basedir) logSeverity overrideProject skipIprFlag licenseOnlyScan fileFilters apiOpts metadata binaryPaths = withLogQueue logSeverity $ \queue -> runLogger queue $ do
+  let scanType = ScanType (fromFlag SkipIPRScan skipIprFlag) (fromFlag LicenseOnlyScan licenseOnlyScan)
+  let wigginsOpts = generateWigginsOpts basedir logSeverity overrideProject scanType fileFilters apiOpts metadata
 
-  -- Run IPR and Sherlock CLIs concurrently
-  logDebug . pretty $ "[All] Running scan on directory " ++ show basedir
-  logDebug . pretty $ "[All] Scan ID is " <> scanId
-  logDebug "[All] Running IPR and Sherlock scans in parallel"
-  logDebug "[Sherlock] Starting Sherlock scan"
-
-  let sherlockOpts = SherlockOpts basedir scanId sherlockClientToken sherlockClientId sherlockUrl sherlockOrgId locator projectRevision vpsOpts'
+  logDebug "Running VPS plugin scan"
   let runIt = runLogger queue . runDiagnostics . runExecIO
-  (iprResult, sherlockResult) <- sendIO $ concurrently
-                (runIt $ runIPRScan basedir apiOpts scanId binaryPaths syOpts vpsOpts')
-                (runIt $ runSherlockScan binaryPaths sherlockOpts)
 
-  case (iprResult, sherlockResult) of
-    (Right _, Right _) -> logDebug "[All] Scans complete"
-    (Left iprFailure, _) -> do
-      logDebug "[IPR] Failed to scan"
-      logDebug $ renderFailureBundle iprFailure
-      sendIO exitFailure
-    (_, Left sherlockFailure) -> do
-      logDebug "[Sherlock] Failed to scan"
-      logDebug $ renderFailureBundle sherlockFailure
+  wigginsResult <- sendIO $ runIt (runWiggins binaryPaths wigginsOpts)
```

Comment:
Since we aren't using `concurrently` anymore, we can just use the current effect wrappers.  We can remove the entire `runIt` line, and this can probably become
```haskell
wigginsResult <- sendIO . runExecIO $ runWiggins binaryPaths wigginsOpts
```

Reply:
This doesn't seem to result in a `(Right, Left)` tuple, which I think we need for rendering the failure right?

Reply:
Any errors in the execution should be "thrown" when we run the effect at the `runExecIO` call.  That will raise any errors into the `Diagnostics` effect we're currently working in (this function has the constraint `Has Diagnostics sig m`), and therefore has to be handled by the caller.  Since `runWiggins` actually returns `m ()`, this line could probably just be `runExecIO $ runWiggins binaryPaths wigginsOpts`, without `sendIO` or `result <-`.  Actually throwing the error should be lower in the call chain, probably in `execWiggins`, but reporting it to console is done in `scanMain` in this file.

Reply:
Ok, that makes sense. Made this change and the one in your other outstanding comment and both error and non-error cases work great! Thanks!